com.unity.render-pipelines.core/Editor/Lighting/LightAnchorEditor.cs

```diff
@@ -0,0 +1,648 @@
+            ToolManager.activeToolChanged += EditorToolsOnactiveToolChanged;
+
+            // Get a reference to the GameView
+            // TODO: This currently only works with a single Game View
+            var gameViews = Resources.FindObjectsOfTypeAll<EditorWindow>()
+                .Where(w => w.GetType().FullName == "UnityEditor.GameView")
+                .ToArray();
+            var gameView = gameViews.First();
+
+            // We create an invisible VisualElement that gets overlayed onto the Game View to intercept mouse clicks
+            // when the editor tool is active
+            m_GameViewRootElement = gameView.rootVisualElement;
+            m_ClickCatcher = new VisualElement();
+            m_ClickCatcher.style.flexGrow = 1;
+            m_ClickCatcher.RegisterCallback<MouseDownEvent>(OnMouseDown);
+
+            // Determine if the editor tool is already active (this helps keep our state when swapping between
+            // different light objects directly, or after an assembly reload)
+            var anchorTools = Resources.FindObjectsOfTypeAll<LightAnchorEditorTool>()
+                .ToArray();
+            if (anchorTools.Length > 0)
+            {
+                var anchorTool = anchorTools[0];
+                if (anchorTool != null)
+                {
+                    EnableClickCatcher(m_EnableClickCatcher);
+                }
+            }
+        }
+
+        void EditorToolsOnactiveToolChanged()
+        {
+            // When the active editor tool changes we need to either enable or disable our click catcher
+            EnableClickCatcher(ToolManager.activeToolType == typeof(LightAnchorEditorTool));
+        }
+
+        void OnDisable()
+        {
+            // cleanup our click catcher element
+            m_ClickCatcher?.RemoveFromHierarchy();
+
+            // unsubscribe from EditorTool event
+            ToolManager.activeToolChanged -= EditorToolsOnactiveToolChanged;
+        }
+
+        void OnMouseDown(MouseDownEvent evt)
+        {
+            // On right mouse button we cast a ray and then move the light anchor to the hit point
+            if (evt.button == 1)
+            {
+                // The mouse position reported by the event is not in the same space as Screen Space.  The Y axis is flipped
+                // Also we need to account for the toolbar.  I couldn't find any way to get an accurate number so this is a best
+                // guess for now
+
+                // Useful to uncomment for debugging
+                //var newMousePos = m_ClickCatcher.WorldToLocal(evt.mousePosition);
+                //var toolbarHeight = 21f;
+                //var mousePos = new Vector3(newMousePos.x, Screen.height - (newMousePos.y + toolbarHeight), 0f);
+                //var ray = Camera.main.ScreenPointToRay(Event.current.mousePosition);
+                //Debug.DrawLine(ray.origin, ray.origin + ray.direction * 100f, Color.yellow, 10f);
+            }
+        }
```

Comment:
At this rate, perhaps remove the method no? (or comment it fully)

Reply:
[Fixed]